authority/tls.go

```diff
@@ -340,7 +352,28 @@ func (a *Authority) Revoke(ctx context.Context, revokeOpts *RevokeOptions) error
 
 	if provisioner.MethodFromContext(ctx) == provisioner.SSHRevokeMethod {
 		err = a.db.RevokeSSH(rci)
-	} else { // default to revoke x509
+	} else {
+		// Revoke an X.509 certificate using CAS. If the certificate is not
+		// provided we will try to read it from the db.
+		var revokedCert *x509.Certificate
+		if revokeOpts.Crt != nil {
+			revokedCert = revokeOpts.Crt
+		} else if rci.Serial != "" {
+			revokedCert, _ = a.db.GetCertificate(rci.Serial)
```

Comment:
Assuming we need to throw an error here?

Reply:
if GetCertificate fails

Reply:
The certificate is might be only required by specific CAS implementations. CloudCAS requires it, and it will throw an error if the the certificate is nil, but SoftCAS doesn't care.

Reply:
Add comment in d644274